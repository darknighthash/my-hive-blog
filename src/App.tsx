import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Home from './pages/Home';
import Post from './pages/Post';
import Header from './components/Header';
import Footer from './components/Footer';

const App: React.FC = () => {
  return (
    <Router>
      <div className="flex flex-col min-h-screen">
        <Header />
        <main className="flex-grow">
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/:author/:permlink" element={<Post />} />
          </Routes>
        </main>
        <Footer />
      </div>
    </Router>
  );
};

export default App;