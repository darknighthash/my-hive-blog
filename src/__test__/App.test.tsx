// src/__tests__/App.test.tsx
import { render, screen ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../App';

describe('App Component', () => {
    // 测试渲染
    test('renders Vite and React logos', () => {
      render(<App />);
  
      // 检查 Vite Logo 是否渲染
      const viteLogo = screen.getByAltText('Vite logo');
      expect(viteLogo).toBeInTheDocument();
  
      // 检查 React Logo 是否渲染
      const reactLogo = screen.getByAltText('React logo');
      expect(reactLogo).toBeInTheDocument();
    });
  
    test('renders the title "Vite + React"', () => {
      render(<App />);
  
      // 检查标题是否正确显示
      const titleElement = screen.getByText(/Vite \+ React/i);
      expect(titleElement).toBeInTheDocument();
    });
  
    test('renders the initial count as 0', () => {
      render(<App />);
  
      // 检查按钮是否显示初始计数 0
      const buttonElement = screen.getByText(/Count is 0/i);
      expect(buttonElement).toBeInTheDocument();
    });
  
    // 测试交互
    test('increments count when button is clicked', () => {
      render(<App />);
  
      // 找到按钮并点击
      const buttonElement = screen.getByText(/Count is 0/i);
      fireEvent.click(buttonElement);
  
      // 检查计数是否增加到 1
      expect(buttonElement.textContent).toBe('Count is 1');
    });
  
    // 快照测试
    test('matches snapshot', () => {
      const { asFragment } = render(<App />);
      expect(asFragment()).toMatchSnapshot();
    });
  });