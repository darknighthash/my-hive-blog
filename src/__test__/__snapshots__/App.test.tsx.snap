// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center min-h-screen bg-lightBg dark:bg-darkBg text-dark dark:text-light"
  >
    <div
      class="flex space-x-4"
    >
      <a
        href="https://vite.dev"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Vite logo"
          class="h-16 w-16 transition-transform duration-300 hover:scale-110"
          src=""
        />
      </a>
      <a
        href="https://react.dev"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="React logo"
          class="h-16 w-16 transition-transform duration-300 hover:scale-110"
          src=""
        />
      </a>
    </div>
    <h1
      class="mt-6 text-4xl font-bold"
    >
      Vite + React
    </h1>
    <div
      class="mt-6 p-6 bg-white dark:bg-gray-800 shadow-lg rounded-lg text-center"
    >
      <button
        class="px-4 py-2 bg-blue-600 text-white rounded-lg shadow-md hover:bg-blue-700 focus:outline-none focus:ring focus:ring-blue-300"
      >
        Count is 0
      </button>
      <p
        class="mt-4 text-gray-600 dark:text-gray-300"
      >
        Edit 
        <code
          class="bg-gray-200 dark:bg-gray-700 rounded px-1"
        >
          src/App.tsx
        </code>
         and save to test HMR
      </p>
    </div>
    <p
      class="mt-6 text-sm text-gray-500 dark:text-gray-400"
    >
      Click on the Vite and React logos to learn more
    </p>
  </div>
</DocumentFragment>
`;
